packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1026,6 +1026,35 @@ abstract class TextInputClient {
   void connectionClosed();
 }
 
+/// An interface to receive granular information from [TextInput].
+///
+/// See also:
+///
+///  * [TextInput.attach]
+///  * [TextInputConfiguration], to opt-in to receive [TextEditingDelta]'s from
+///    the platforms [TextInput] you must set [TextInputConfiguration.enableDeltaModel]
+///    to true.
+abstract class DeltaTextInputClient extends TextInputClient {
+  /// Requests that this client update its editing state by applying the deltas
+  /// received from the engine.
+  ///
+  /// The list of [TextEditingDelta]'s are treated as changes that will be applied
+  /// to the client's editing state. A change is any mutation to the raw text
+  /// value, or any updates to the selection and/or composing region.
+  ///
+  /// Here is an example of what implementation of this method could look like:
+  ///
+  /// @override
```

Comment:
This example should probably be a `{@tool snippet}` so that it's formatted nicely on the docs website.

Reply:
Thanks!